src/goto-programs/remove_virtual_functions.cpp

```diff
@@ -322,13 +384,99 @@ void remove_virtual_functionst::get_functions(
 
   // iterate over all children, transitively
   std::set<irep_idt> visited;
-  get_child_functions_rec(
+  get_child_functions(
     class_id,
     root_function.symbol_expr,
     component_name,
     functions,
     visited);
 
+  const auto find_empty = [&](const dispatch_table_entryt &entry) {
+    return entry.symbol_expr == symbol_exprt();
+  };
+  bool empty_found =
+    std::any_of(functions.begin(), functions.end(), find_empty);
+
+  if(empty_found && !functions.empty())
+  {
+    std::cout << "problematic dispatch table\n";
+    for(const auto &f : functions)
+      std::cout << f.class_id << ":" << f.symbol_expr.pretty() << " ";
+    std::cout << std::endl;
+
+    // with a non-empty virtual dispatch table but only empty function
```

Comment:
The condition says *any* class fails to provide a definition, but the comment says *all*